File: ui/src/types/BaneType.ts
export enum BaneType {
    BESTEFAR_BANEN = "BESTEFAR_BANEN",
    FRODE_SPESIAL = "FRODE_SPESIAL",
    UTEN_NAVN = "UTEN_NAVN",
    KORTESTE_VEIEN = "KORTESTE_VEIEN",
    UTENFOR_BANEN = "UTENFOR_BANEN",
    I_DEPO = "I_DEPO",
}


export function parseBaneType(value: string): BaneType | undefined {
    return (Object.values(BaneType) as string[]).includes(value) ? (value as BaneType) : undefined;
}

File: src/main/java/no/vejmon/dommern/bane/BaneType.java
package no.vejmon.dommern.bane;

import lombok.Getter;
import lombok.Setter;

@Getter
public enum BaneType {
    BESTEFAR_BANEN(true),
    FRODE_SPESIAL(true),
    UTEN_NAVN(true),
    KORTESTE_VEIEN(true),
    UTENFOR_BANEN(false),
    I_DEPO(false);

    private final boolean enabled;

    @Setter
    private Integer gpioPin;
    @Setter
    private Integer soundNumber;

    BaneType(boolean enabled) {
        this.enabled = enabled;
    }
}